openhcl/underhill_mem/src/mapping.rs

```diff
@@ -599,4 +611,50 @@ unsafe impl GuestMemoryAccess for GuestMemoryMapping {
                 .map(|bitmap| bitmap.access_bitmap())
         }
     }
+
+    fn page_fault(
+        &self,
+        address: u64,
+        _len: usize,
+        _write: bool,
+        bitmap_failure: bool,
+    ) -> guestmem::PageFaultAction {
+        let err = {
+            if bitmap_failure {
+                if !self
+                    .valid_memory
+                    .as_ref()
+                    .expect("all backings with bitmaps should have a GuestValidMemory")
+                    .check_valid(address / PAGE_SIZE as u64)
+                {
+                    if self.iova_offset == Some(0) {
```

Comment:
Let's have an explicit boolean or enum that says what kind of mapping we're in rather than derive it from iova_offset.